tests/test_cases/test_cocotb/test_scheduler.py

```diff
@@ -1051,3 +1054,46 @@ async def coro() -> None:
         t = await cocotb.start(coro(), name="baz")
     assert t.get_name() == "baz"
     t.cancel()
+
+
+@cocotb.test(expect_error=RuntimeError)
+async def test_start_again_finished_task(_: object) -> None:
+    async def coro():
+        await Timer(1, "ns")
+
+    a = cocotb.start_soon(coro())
+    await a
+
+    cocotb.start_soon(a)
```

Comment:
Can we make this more specific?
```suggestion
@cocotb.test
async def test_start_again_finished_task(_: object) -> None:
    async def coro() -> None:
        await Timer(1, "ns")

    a = cocotb.start_soon(coro())
    await a

    with pytest.raises(RuntimeError):
        cocotb.start_soon(a)
```